src/Arcus.Security.Core/Providers/ConfigurationSecretProvider.cs

```diff
@@ -46,7 +51,10 @@ public Task<Secret> GetSecretAsync(string secretName)
         /// <exception cref="T:Arcus.Security.Core.SecretNotFoundException">The secret was not found, using the given name</exception>
         public Task<string> GetRawSecretAsync(string secretName)
         {
-            Guard.NotNullOrWhitespace(secretName, nameof(secretName), "Requires a non-blank secret name to look up the secret configuration value");
+            if (string.IsNullOrWhiteSpace(secretName))
+            {
+                throw new ArgumentNullException(nameof(secretName), "Requires a non-blank secret name to look up the secret configuration value");
```

Comment:
Should'nt this be an `ArgumentException` if we want to be consistent ?
```suggestion
                throw new ArgumentException("Requires a non-blank secret name to look up the secret configuration value", nameof(secretName));
```